eng/targets/Imports.targets

```diff
@@ -237,4 +237,51 @@
       </TargetFrameworkMonikerAssemblyAttributeText>
     </PropertyGroup>
   </Target>
+
+  <Target Name="_CalculateSubstituteVariablesInApplicationManifestInputsOutputs">
+    <PropertyGroup>
+      <_OriginalApplicationManifestPath>$([System.IO.Path]::Combine('$(MSBuildProjectDirectory)', '$(ApplicationManifest)'))</_OriginalApplicationManifestPath>
+      <_UpdatedApplicationManifestPath>$(IntermediateOutputPath)$(AssemblyName).exe.manifest</_UpdatedApplicationManifestPath>
```

Comment:
Do we always produce a `.exe`? on netcore I'd assume we produce a `csc.dll`, does the manifest name need to match in that case?

Reply:
Not 100% clear, but worth noting that long paths are enabled by default in netcore.

Reply:
Do you have any thoughts on this @tmat?

Reply:
> long paths are enabled by default in netcore.

Presumably because we launch via dotnet.exe, which itself has a manifest that enables long paths? In that case for netcore we wouldn't need the manifests, so seems good.